### PR TITLE
Update autoscaling to use service override

### PR DESCRIPTION
**Pull request by Deydra71**

### CODE REVIEW

File: controllers/aodh_controller.go [hunk 1]
```diff
@@ -332,14 +332,25 @@ func (r *AutoscalingReconciler) reconcileNormalAodh(
 	for endpointType, data := range aodhEndpoints {
 		endpointTypeStr := string(endpointType)
 		endpointName := autoscaling.ServiceName + "-" + endpointTypeStr
-		svcOverride := instance.Spec.Aodh.Override.Service
-		if svcOverride == nil {
-			svcOverride = &service.RoutedOverrideSpec{}
+
+		// Make sure the Service map in the Override is initialized
+		if instance.Spec.Aodh.Override.Service == nil {
```
Comment (stuggi) [inline controllers/aodh_controller.go:337]: wouldn't it be ok to do it like in the other operators? e.g. https://github.com/openstack-k8s-operators/ironic-operator/blob/main/controllers/ironicapi_controller.go#L427

Comment (Deydra71) [reply controllers/aodh_controller.go:337]: I see, on it, thanks!

**Comment by Deydra71** [comment]
The RabbitMQ image pull error is happening here as well:

`    case.go:364: failed in step 1-deploy
    case.go:366: services "ceilometer" not found
    case.go:366: pods "ceilometer-0" not found
    case.go:366: statefulsets.apps "ceilometer" not found
    case.go:366: services "aodh-internal" not found
    case.go:366: services "aodh-public" not found
    case.go:366: pods "aodh-0" not found
    case.go:366: statefulsets.apps "aodh" not found
`
....
`$ oc get pods -n telemetry-kuttl-tests
rabbitmq-server-0    0/1     Init:ImagePullBackOff   0          47s
`

**Comment by Deydra71** [comment]
recheck

**Pull request merged**